ldap/servers/slapd/regex.c

```diff
@@ -36,18 +34,20 @@ struct slapi_regex_handle
  * \warning The regex handler should be released by slapi_re_free().
  */
 Slapi_Regex *
-slapi_re_comp(const char *pat, const char **error)
+slapi_re_comp(const char *pat, char **error)
 {
     Slapi_Regex *re_handle = NULL;
-    pcre *re = NULL;
-    const char *myerror = NULL;
-    int erroffset;
+    pcre2_code *re = NULL;
+    int32_t myerror;
+    PCRE2_SIZE erroffset;
+    PCRE2_UCHAR errormsg[CPRE_ERR_MSG_SIZE];
 
-    re = pcre_compile(pat, 0, &myerror, &erroffset, NULL);
-    if (error) {
-        *error = myerror;
-    }
-    if (re) {
+    re = pcre2_compile((PCRE2_SPTR)pat, strlen(pat), 0,
+                       &myerror, &erroffset, NULL);
+    if (re == NULL) {
```

Comment:
> In case the caller did not initialize 'error = NULL', it would be safer to systematically do '*error = NULL'.

I reworked it, let me know if you are okay with it.